sklearn/metrics/_classification.py

```diff
@@ -2567,9 +2569,18 @@ def log_loss(
     ...          [[.1, .9], [.9, .1], [.8, .2], [.35, .65]])
     0.21616...
     """
+    if eps == "auto":
+        if "numpy" in str(type(y_pred)) and "float" in str(y_pred.dtype):
+            eps = np.finfo(y_pred.dtype).eps
+        else:
+            warnings.warn(
+                "eps set to 1e-15 as y_pred is not a numpy float array",
+                DataConversionWarning,
+            )
+            eps = 1e-15
```

Comment:
We don't need the `DataConversionWarning` here. We can simplify by only setting `eps` as below.